asserts/header_checks.go

```diff
@@ -33,6 +33,14 @@ func checkMandatory(headers map[string]string, name string) (string, error) {
 	if !ok {
 		return "", fmt.Errorf("%q header is mandatory", name)
 	}
+	return value, nil
+}
+
+func checkNotEmpty(headers map[string]string, name string) (string, error) {
+	value, err := checkMandatory(headers, name)
```

Comment:
This is a slightly dangerous change to make, because checkMandatory changed its meaning without breaking existing code. So reviewing the branch becomes impossible, and even merging the branch becomes a potential source of serious bugs because if someone else commits code meanwhile this would go in cleanly without perceiving.

To avoid that problem, can we please rename checkMandatory to checkExists? This will catch any issues and make the review more meaningful.